### PR TITLE
feat: support custom containerd config dir

### PR DESCRIPTION
**Pull request by anjia0532**
Signed-off-by: 赵安家 <anjia0532@gmail.com>

### CODE REVIEW

**Comment by jim3ma** [comment]
Thanks for this PR.

Something TODO:

1. search for `/etc/containerd`, there are others to replace.
2. Run `pre-commit` check: https://github.com/dragonflyoss/helm-charts/blob/main/charts/README.md#install-pre-commit

File: charts/dragonfly/values.yaml [hunk 1]
```diff
@@ -91,6 +91,9 @@ containerRuntime:
     # if not, init container will just add the dfdaemon.config.proxy.registryMirror.url for registry mirror.
     # When configPath is true, init container will inject config_path=/etc/containerd/certs.d and configure all registries.
     injectConfigPath: false
+    # -- Custom config path directory ,default is /etc/containerd/
```
Comment (jim3ma) [inline charts/dragonfly/values.yaml:94]: `directory ,default` -> `directory, default`

File: charts/dragonfly/values.yaml [hunk 2]
```diff
@@ -91,6 +91,9 @@ containerRuntime:
...
+    # e.g. rke2 generator config path is /var/lib/rancher/rke2/agent/etc/containerd/config.toml, docs: https://github.com/rancher/rke2/blob/master/docs/advanced.md#configuring-containerd
+    configPathDir: /host/etc/containerd/
```
Comment (jim3ma) [inline charts/dragonfly/values.yaml:96]: `configPathDir` should not contain `/host` prefix. Prepend `/host` in `charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml`

**Review by jim3ma — CHANGES_REQUESTED** [review]

File: charts/dragonfly/values.yaml (hunk 1 above)
Comment (anjia0532) [reply charts/dragonfly/values.yaml]: Done

File: charts/dragonfly/values.yaml (hunk 2 above)
Comment (anjia0532) [reply charts/dragonfly/values.yaml]: Done

**Comment by anjia0532** [comment]
Please review again

File: charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml [hunk 3]
```diff
@@ -300,7 +300,7 @@ spec:
         - /bin/sh
         - -cx
         - |-
-          etcContainerd=/host/etc/containerd
+          etcContainerd={{ .Values.containerRuntime.containerd.configPathDir }}
```
Comment (jim3ma) [inline charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml:303]: `etcContainerd={{ .Values.containerRuntime.containerd.configPathDir }}` ->
`etcContainerd=/host{{ .Values.containerRuntime.containerd.configPathDir }}`

**Review by jim3ma — CHANGES_REQUESTED** [review]
Update this line in `charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml`:
```
config_path = "/etc/containerd/certs.d"
```

File: charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml (hunk 3 above)
Comment (anjia0532) [reply charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml:303]: Maybe add `dfdaemon.extraVolumeMounts` to https://github.com/dragonflyoss/helm-charts/blob/01207e893cac17389206898ed9c98108b6eabca0/charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml#L405-L407 ?

```

        {{- if .Values.dfdaemon.extraVolumeMounts }}
        {{- toYaml .Values.dfdaemon.extraVolumeMounts | nindent 8 }}
        {{- end }}
```

Comment (jim3ma) [reply charts/dragonfly/templates/dfdaemon/dfdaemon-daemonset.yaml:303]: I think just replace `/etc` to `{{ .Values.containerRuntime.containerd.configPathDir }}` in volumes  and update `/host/etc ` to `/host{{ .Values.containerRuntime.containerd.configPathDir }}` in volumeMounts. BTW, change `name: etc` to `containerd` or other name.

**Review by jim3ma — APPROVED** [review]
LGTM

**Pull request merged**